app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/AccountController.scala

```diff
@@ -101,23 +105,46 @@ class AccountController @Inject() (
     } yield (ej, uj)
 
   private def renderAccountPage(undertaking: Undertaking)(implicit r: AuthenticatedEscRequest[AnyContent]) = {
-    val currentTime = timeProvider.today
+    implicit val eori = r.eoriNumber
+    val currentDay = timeProvider.today
 
-    val isTimeToReport = ReportReminderHelpers.isTimeToReport(undertaking.lastSubsidyUsageUpdt, currentTime)
+    val isTimeToReport = ReportReminderHelpers.isTimeToReport(undertaking.lastSubsidyUsageUpdt, currentDay)
     val dueDate = ReportReminderHelpers.dueDateToReport(undertaking.lastSubsidyUsageUpdt).map(_.toDisplayFormat)
-    val isOverdue = ReportReminderHelpers.isOverdue(undertaking.lastSubsidyUsageUpdt, currentTime)
-
-    if (undertaking.isLeadEORI(r.eoriNumber)) {
-      Ok(
-        leadAccountPage(
-          undertaking,
-          undertaking.getAllNonLeadEORIs().nonEmpty,
-          isTimeToReport,
-          dueDate,
-          isOverdue
-        )
-      )
-    } else Ok(nonLeadAccountPage(undertaking))
+    val isOverdue = ReportReminderHelpers.isOverdue(undertaking.lastSubsidyUsageUpdt, currentDay)
+
+    if (undertaking.isLeadEORI(eori)) {
+      val result = for {
+        nilReturnJourney <- store
+          .get[NilReturnJourney]
+          .toContext
+          .orElse(store.put[NilReturnJourney](NilReturnJourney()).toContext)
+        updatedJourney <-
+          if (nilReturnJourney.nilReturnCounter == 1 || nilReturnJourney.nilReturnCounter == 2)
```

Comment:
It's not clear what 1 and 2 mean here. Perhaps extract this check into a function with a meaningful name and consider providing named constants for the values if required.